coredocument/coredocument.go

```diff
@@ -16,6 +16,41 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
+const (
+	CDRootField       = "cd_root"
+	DataRootField     = "data_root"
+	DocumentTypeField = "document_type"
+	SignaturesField   = "signatures"
+	SigningRootField  = "signing_root"
+)
+
+var compactProperties = map[string][]byte{
+	CDRootField:       {0, 0, 0, 7},
+	DataRootField:     {0, 0, 0, 5},
+	DocumentTypeField: {0, 0, 0, 100},
+	SignaturesField:   {0, 0, 0, 6},
+	SigningRootField:  {0, 0, 0, 10},
+}
+
+func NewDefaultTree(salts *proofs.Salts) *proofs.DocumentTree {
+	return NewDefaultTreeWithPrefix(salts, "")
+}
+
+func NewDefaultTreeWithPrefix(salts *proofs.Salts, prefix string) *proofs.DocumentTree {
+	var prop proofs.Property
```

Comment:
Technically we also have to prefix the compact version of the properties so we'll need to have a byte and string prefix. I think it's fine to implement this in a later PR but it's something we should think about. Perhaps we actually can pass in a Property as a prefix instead.

Reply:
True, I will add another entry in the map for our prefixes used as a "reserved" prefix.

Reply:
The lengthProp method needs to add as well the compact property for `length`: https://github.com/centrifuge/precise-proofs/blob/13d3af957299c614237c42cdc331f7acd7c7d201/proofs/property.go#L127-L133